packages/react-components/react-table/src/contexts/tableHeaderContext.ts

```diff
@@ -0,0 +1,8 @@
+import * as React from 'react';
+
+const tableHeaderContext = React.createContext<string | undefined>(undefined);
+
+const tableHeaderContextDefaultValue = '';
+
+export const TableHeaderContextProvider = tableHeaderContext.Provider;
+export const useIsInTableHeader = () => React.useContext(tableHeaderContext) === tableHeaderContextDefaultValue;
```

Comment:
Is the provider part of the public API for `Table`?

It feels a little odd for an empty string to mean "this row is in the header", particularly where the value is set on the provider.

Reply:
It could also be a boolean I guess, the value itself isn't that important but rather it should be an indicator that it is inside the header

Reply:
Is it something we will expose to users or will it only ever be set internally by the component?

After thinking about it some more, if it's only internal I'm not really concerned about this.